### PR TITLE
Expose more Asynchbase properties, refactor AsynchbaseSchemaService, …

### PR DESCRIPTION
**Pull request by bsura**
…add counter to track no. of metric committed.

### CODE REVIEW

**Comment by bsura** [comment]
@dilipdevaraj-sfdc Please review and merge.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 1]
```diff
@@ -89,17 +93,31 @@
 
     @SLF4JTypeListener.InjectLogger
     private Logger _logger;
-    private HBaseClient _client;
+    private final HBaseClient _client;
+    private final boolean syncPut;
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Foe consistency sake can you change private member to _syncPut

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 2]
```diff
@@ -528,7 +501,79 @@ public Void call(Exception arg) throws Exception {
             }
         }
     }
+    
+    
+    private void _ensureTableWithColumnFamilyExists(byte[] table, byte[] family) {
+		final CountDownLatch latch = new CountDownLatch(1);
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Can you indent the code

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java [hunk 3]
```diff
@@ -528,7 +501,79 @@ public Void call(Exception arg) throws Exception {
...
+    	_put(tableName, rowKeyStr);
+    }
+    
+    private void _putWithTag(Metric metric, Entry<String, String> tag, String tableName) {
+    	String rowKeyStr = _constructRowKey(metric.getNamespace(), metric.getScope(), metric.getMetric(), tag.getKey(), 
+    			tag.getValue(), TableType.fromTableName(tableName));
+    	_put(tableName, rowKeyStr);
+    }
+
+	private void _put(String tableName, String rowKeyStr) {
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Correct indent here too
You could use Ctrl+I in eclipse

Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Or could use Jalopy locally

Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: @bsura  Done

**Comment by saaja-sfdc** [comment]
@bsura Are there updates that need to be added to this PR?  I'd like to merge and then integrate back onto develop.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java (hunk 1 above)
Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/AsyncHbaseSchemaService.java]: Done.

**Comment by bsura** [comment]
@saaja-sfdc I pushed one last commit to change variable name as Dilip suggested. It can be merged once the tests pass.

**Pull request merged**